### PR TITLE
Openstack keystone v3 support

### PR DESCRIPTION
**Pull request by epol**
Added support for the new keystone v3 API, the old v2.0 APIs are still available, the v3 can be selected with a version option.
Only password login is implemented here for API v3, so token login is still missing.

Keystone API v2.0 has been deprecated years ago and many new openstack installations don't support them while API v3 is the current version of the APIs. See also issue #2179

With this patch API v3 support is added, but the default is still v2.0 so backward compatibility is preserved.

### CODE REVIEW

**Comment by epol** [comment]
In my opinion most warnings in the quality review aren't correct, for example https://app.codacy.com/app/kenneth/duplicati/file/16136819852/issues/source?bid=3742986&fileBranchId=6972408#l240 says the 'url' is not used, but it's used here: https://github.com/epol/duplicati/blob/master/Duplicati/Library/Backend/OpenStack/OpenStackStorage.cs#L442

File: Duplicati/Library/Backend/OpenStack/OpenStackStorage.cs [hunk 1]
```diff
@@ -122,6 +227,32 @@ public OpenStackAuthRequest(string tenantname, string username, string password,
...
         }
 
+        private class Keystone3AuthResponse
+        {
+            public TokenClass token { get; set; }
+
+            public class EndpointItem
+            {
+                // 'interface' is a reserved keyword, so we need this decorator to map it
+                [JsonProperty(PropertyName = "interface")]
```
Comment (kenkendk) [inline Duplicati/Library/Backend/OpenStack/OpenStackStorage.cs]: You can also use `public string @interface { get; set; }`

**Comment by kenkendk** [comment]
Great work! Full-stack fix :)

The Codacity setup is not yet refined enough. We can disable some of the rules.

**Comment by epol** [comment]
Codacy has been helpful to find some forgotten code, so maybe it's better to have some extra warning and take them with a grain of salt ;)

Thank you for accepting the PR and for the great work on this software!

**Pull request merged**